migration/migrator_validate.go

```diff
@@ -0,0 +1,138 @@
+package migration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+func (m *StateMigrator) ValidateMigratedState(mptRoot common.Hash, zkRoot common.Hash) error {
+	var accounts atomic.Uint64
+	var slots atomic.Uint64
+
+	eg, cCtx := errgroup.WithContext(context.Background())
```

Comment:
How about just adding ctx to this function's input and pass context.Background() from the validation goroutine after MPT time?
```suggestion
	eg, cCtx := errgroup.WithContext(ctx)
```